ethcore/src/client/mod.rs

```diff
@@ -65,20 +65,48 @@ pub trait BlockChainClient : Sync + Send {
 	/// Get block total difficulty.
 	fn block_total_difficulty(&self, id: BlockID) -> Option<U256>;
 
-	/// Get address nonce.
-	fn nonce(&self, address: &Address) -> U256;
+	/// Attempt to get address nonce at given block.
+	/// May not fail on BlockID::Latest.
+	fn nonce(&self, address: &Address, id: BlockID) -> Option<U256>;
+
+	/// Get address nonce at the latest block's state.
+	fn nonce_latest(&self, address: &Address) -> U256 {
```

Comment:
That is how I had it initially, but I found that the breakage caused by the change in signature was actually very small in reality. `State::nonce` and `Miner::nonce` are unchanged, and this is where most code actually looks for nonces.